### ISSUE TITLE
Point to line snapping isn't sensitive enough

### ISSUE THREAD

**Issue by aronkvh** [facebook/Rapid#863]
### Description

When I'm drawing 2 buildings that are connected - a  point with a line - iD snaps the point to the line, but RapiD 2 beta only snaps if I zoom in really really close and place it exactly on the area outline

### Version

2.0.0-beta.0

### What browser are you seeing the problem on? What version are you running?

Firefox 111.0.1

### The OS you're using

Linux - KDE Neon

### The browser URL at the time you encountered the bug

https://mapwith.ai/rapid-v2-beta#background=Bing&datasets=fbRoads,msBuildings&disable_features=boundaries&map=20.40/47.03014/19.77862&offset=14.71,-4.24

**Comment by bhousel** [comment]
Yeah I have noticed this too.. I think we are having issues detecting exactly which line is being hit by the user's cursor (the line being snapped to vs the line being drawn).  Point snapping works great, but line snapping is iffy.

I think we can fix this by drawing line segments as separate Pixi graphics, that way we can hit detect them separately as well (this will help with culling too, and they will anyway get batched together when it comes time to prepare the WebGL buffers).  

Back in Rapid v1, we used to generate separate invisible hit targets for each line segment - that is how we were able to "nope" 🚫 cursor out some segments of the lines and prevent snapping to them.
